beacon_node/http_api/src/lib.rs

```diff
@@ -1875,35 +1873,49 @@ pub fn serve<T: BeaconChainTypes>(
             },
         );
 
-    let post_beacon_pool_attestations_v2 = beacon_pool_path_v2
-        .clone()
-        .and(warp::path("attestations"))
-        .and(warp::path::end())
-        .and(warp_utils::json::json())
-        .and(network_tx_filter.clone())
-        .and(reprocess_send_filter)
-        .and(log_filter.clone())
-        .then(
-            |task_spawner: TaskSpawner<T::EthSpec>,
-             chain: Arc<BeaconChain<T>>,
-             attestations: Vec<SingleAttestation>,
-             network_tx: UnboundedSender<NetworkMessage<T::EthSpec>>,
-             reprocess_tx: Option<Sender<ReprocessQueueMessage>>,
-             log: Logger| async move {
-                let attestations = attestations.into_iter().map(Either::Right).collect();
-                let result = crate::publish_attestations::publish_attestations(
-                    task_spawner,
-                    chain,
-                    attestations,
-                    network_tx,
-                    reprocess_tx,
-                    log,
-                )
-                .await
-                .map(|()| warp::reply::json(&()));
-                convert_rejection(result).await
-            },
-        );
+    let post_beacon_pool_attestations_v2 =
+        beacon_pool_path_v2
+            .clone()
+            .and(warp::path("attestations"))
+            .and(warp::path::end())
+            .and(warp_utils::json::json::<Value>())
+            .and(optional_consensus_version_header_filter)
+            .and(network_tx_filter.clone())
+            .and(reprocess_send_filter.clone())
+            .and(log_filter.clone())
+            .then(
+                |task_spawner: TaskSpawner<T::EthSpec>,
+                 chain: Arc<BeaconChain<T>>,
+                 payload: Value,
+                 fork_name: Option<ForkName>,
+                 network_tx: UnboundedSender<NetworkMessage<T::EthSpec>>,
+                 reprocess_tx: Option<Sender<ReprocessQueueMessage>>,
+                 log: Logger| async move {
+                    let Ok(attestations) =
+                        crate::publish_attestations::deserialize_attestation_payload::<T>(
+                            payload, fork_name, &log,
+                        )
+                    else {
+                        return warp::reply::with_status(
+                            warp::reply::json(&"Unable to deserialize request body".to_string()),
+                            eth2::StatusCode::BAD_REQUEST,
+                        )
+                        .into_response();
+                    };
```

Comment:
We are burying the specific error here, but it might be good to at least log it if we don't return it in the response.

Reply:
Added a warning log for this here: 6eb0451237a00ecf61f9202a5b31dd312f3a446d